### PR TITLE
Add whitelist Travis task

### PR DESCRIPTION
**Pull request by koops76**

### CODE REVIEW

**Comment by koops76** [comment]
Test failure is expected.

**Comment by koops76** [comment]
Test success is expected now. 😁

**Comment by koops76** [comment]
@Hainish This is ready for merge.

**Comment by cschanaj** [comment]
@koops76 does this implies that we will have to remove an existing entry manually every time we update the items in the list?

P.S. I am not really good at `git`, please correct me if I am wrong :smile:

**Comment by koops76** [comment]
@cschanaj Yes, it's not done automatically. Also keeping the alphabetic order is now necessary for the tests to pass.

**Comment by cschanaj** [comment]
@koops76 This is a good idea since it will keep the white-list synced. However, there is a few points I am concerned about

- Some contributors, like @Bisaloo and I have been working on the items in the white-list. This PR will reinforce us to run the cleanup script whenever there is a change to the white-listed item. IMHO, this is sort of troublesome if the respective changes are massive.

- Suggestion: there should be an error message suggesting the contributor to run the cleanup script when Travis failed `whitelist`. (enhancement)

- Suggestion: if possible, maybe we should check that _ONLY the changed files are added/ removed from the whitelist_. This will avoid unnecessary or possibly malicious whitelisting. (optional)

I am open to merging this with or without the aforementioned changes. thanks.

Possibly Related: #11597, #11560

**Comment by koops76** [comment]
@cschanaj
1. You can simply ignore `whitelist` test failure until the rest of the PR is finished.

File: test/travis.sh [hunk 1]
```diff
@@ -87,4 +88,11 @@ if [ "$RULESETS_CHANGED" ]; then
   fi
 fi
 
+if $RULESETS_CHANGED || $WHITELIST_CHANGED; then
+  if [ "$TEST" == "whitelist" ]; then
+    utils/ruleset-coverage-whitelist-cleanup.sh
+    [ `git diff --name-only | wc -l` -eq 0 ]
```
Comment (cschanaj) [inline test/travis.sh]: @koops76 here? with a brief message I guess.

**Comment by Hainish** [comment]
Adding more tests to `.travis.yml` increases the overall running time of Travis.  In this case, there's no concrete benefit - your tests will fail if the whitelist is out of sync, and that adds a development burdon.  For the longest time we had no cleanup script at all, and that was mostly okay.  The periodic run of the cleanup script, upon every release, is sufficient.

**Comment by koops76** [comment]
@Hainish If the script will be modified to only check changed entries, will you add it to tests?

**Pull request closed without merging**